cynic-codegen/src/utils.rs

```diff
@@ -0,0 +1,18 @@
+use darling::util::SpannedValue;
+use proc_macro2::Span;
+
+pub trait ExtractString<T> {
+    fn extract_spanned_value(&self, ident: String) -> SpannedValue<String>;
+    fn extract_spanned(&self) -> Span;
+}
+
+impl ExtractString<Option<SpannedValue<String>>> for Option<SpannedValue<String>> {
+    fn extract_spanned_value(&self, ident: String) -> SpannedValue<String> {
+        self.clone()
+            .unwrap_or(SpannedValue::new(ident, Span::call_site()))
+    }
+
+    fn extract_spanned(&self) -> Span {
+        self.clone().unwrap_or(SpannedValue::default()).span()
+    }
+}
```

Comment:
If you take my comment about adding functions to the input structs then this won't be needed.

But some pointers:

1. You definitely don't need that `<T>` generic parameter on this trait.  Imagine you may have copied that pattern from elsewhere in the codebase, but it's not needed here.
2. Generally would avoid using `Span::call_site` where possible (in this case the `ident` would have had a span you could have used if you'd kept it as a `proc_macro2::Ident` rather than a String).
3. It's fairly common practice to name traits like this `OptionSpannedValueExt` or similar to recognise that they're basically a trait used for extending a single type.

Reply:
Thanks